src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go

```diff
@@ -220,21 +216,29 @@ func (sms *StorageMiningSubsystem_I) VerifyElectionPoSt(header block.BlockHeader
 		return false
 	}
 
-	// 2. Verify appropriate randomness
+	// 2. Verify partialTicket values are appropriate
+	if !sms._verifyElection(header, onChainInfo) {
+		return false
+	}
+
+	// verify the partialTickets themselves
+	// 3. Verify appropriate randomness
 	// TODO: fix away from BestChain()... every block should track its own chain up to its own production.
 	randomness := sms._consensus().GetPoStChallengeRand(sms._blockchain().BestChain(), header.Epoch())
-	postRandomnessInput := sector.PoStRandomness(sms._preparePoStChallengeSeed(randomness, header.Miner()))
+	postRandomnessInput := sector.PoStRandomness(sms.PreparePoStChallengeSeed(randomness, header.Miner()))
 
 	postRand := &filcrypto.VRFResult_I{
 		Output_: onChainInfo.Randomness(),
 	}
 
-	if !postRand.Verify(postRandomnessInput, sms.GetWorkerKeyByMinerAddress(header.Miner())) {
+	// TODO: get worker key from minerAddr
+	var workerKey filcrypto.VRFPublicKey
```

Comment:
This is straightforward when you have `sma`, though you might need to add an accessor method.